tests/func/test_metrics.py

```diff
@@ -723,3 +724,32 @@ def _do_show(self, file_name, xpath):
                 self.assertSequenceEqual(ret[branch][file_name], [branch])
             else:
                 self.assertSequenceEqual(ret[branch][file_name], branch)
+
+
+class TestShouldDisplayMetricsEvenIfMetricIsMissing(TestDvc):
+    branch_missing_metric = "missing_metric_branch"
+    metric_file = "metric"
+    metric_file_stage = metric_file + Stage.STAGE_FILE_SUFFIX
+
+    def test(self):
+        self.dvc.scm.checkout(self.branch_missing_metric, create_new=True)
+
+        with open(self.metric_file, "w+") as fd:
+            fd.write("0.5")
+            fd.flush()
+
+        ret = main(["run", "-M", self.metric_file])
+        assert 0 == ret
+
+        self.dvc.scm.add([self.metric_file_stage])
+        self.dvc.scm.commit("commit msg")
+
+        self.dvc.scm.checkout("master")
+
+        with self._caplog.at_level(logging.INFO, logger="dvc"):
+            ret = main(["metrics", "show", "-a"])
+        assert 0 == ret
+        assert (
+            "{}: {}".format(self.metric_file, "COULD NOT READ")
```

Comment:
if you are using "COULD NOT READ" as a metric, then you better use 
```
self.dvc.metrics.show()
```
which will return a dict that you could easily check without messing around with logs.

Reply:
In updated version we warn user about lack of metric file, now caplog is required